### PR TITLE
Bump version bounds and fix deprecation issues

### PR DESCRIPTION
**Pull request by maralorn**
Kinda business as usual. ;-)

### CODE REVIEW

File: src/Data/Dependent/Sum/Orphans.hs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 {-# LANGUAGE TypeOperators #-}
 {-# LANGUAGE UndecidableInstances #-}
 {-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE PatternSynonyms #-}
```
Comment (3noch) [inline src/Data/Dependent/Sum/Orphans.hs]: Please sort.

Comment (maralorn) [reply src/Data/Dependent/Sum/Orphans.hs]: That extension doesn‘t fix anything, so I revert it …

**Comment by maralorn** [comment]
@3noch I am trying to figure out how to fix the build on ghc 8.0.2 but I can't get it to built. The error is always, that `Data.Some` does not export the constructor `Some(Some)`.

Weirdly if I look at the source of `Data.Some` the constructor is exported. There is some CPP involved, but the CPP should have the same effect on all ghcs we test, i.e. only do something different on ghc 8.0.0 and lower.

cf. https://hackage.haskell.org/package/some-1.0.1/docs/src/Data.Some.Newtype.html

**Comment by maralorn** [comment]
TIL: 801 means ghc 8.1, okay, now I know where the error is.

**Comment by maralorn** [comment]
@3noch finally found a fix.

**Review by 3noch — APPROVED** [review]

**Comment by maralorn** [comment]
Is the remaining failure a problem? I am not sure what the problem is.

**Comment by 3noch** [comment]
The remaining failure blocks merge yes. It looks like release.nix just needs to have some deps bumped since it doesn't use any solving to get the dependencies.

**Comment by maralorn** [comment]
Is there anything I can do?
This seems to be compatibility check with reflex platform? 
There are no nix files in this repo.

**Comment by ali-abrar** [comment]
Thanks @maralorn - I pulled your commits into #7 which will be released as 0.3

**Pull request closed without merging**